packages/amplify-util-uibuilder/src/commands/utils/getApiConfiguration.ts

```diff
@@ -0,0 +1,71 @@
+import { printer } from '@aws-amplify/amplify-prompts';
+import { AmplifyStudioClient } from '../../clients';
+import { $TSContext } from '@aws-amplify/amplify-cli-core';
+import { getUiBuilderComponentsPath } from './getUiBuilderComponentsPath';
+import { getCodegenConfig } from 'amplify-codegen';
+import { ApiConfiguration } from 'aws-sdk/clients/amplifyuibuilder';
+import path from 'path';
+
+export function relativeToComponentsPath(importPath: string, context: $TSContext): string {
+  const componentsPath = getUiBuilderComponentsPath(context);
+  return path.relative(componentsPath, importPath).split(path.sep).join('/');
+}
+
+export function getApiConfiguration(studioClient: AmplifyStudioClient, context: $TSContext): ApiConfiguration {
+  if (studioClient.isDataStoreEnabled) {
+    return {
+      dataStoreConfig: {},
+    };
+  }
+
+  if (studioClient.metadata.isGraphQLEnabled) {
+    printer.debug('building graphql config');
+    // attempt to get api codegen info
+    const projectPath = context.exeInfo.localEnvInfo.projectPath;
+    let promptForUpdateCodegen = false;
+
+    try {
+      const codegenConfig = getCodegenConfig(projectPath);
+      const typesPath = codegenConfig.getGeneratedTypesPath();
+      const apiConfiguration = {
+        graphQLConfig: {
+          typesFilePath: (typesPath && relativeToComponentsPath(typesPath, context)) || '',
+          queriesFilePath: relativeToComponentsPath(codegenConfig.getGeneratedQueriesPath(), context),
+          mutationsFilePath: relativeToComponentsPath(codegenConfig.getGeneratedMutationsPath(), context),
+          subscriptionsFilePath: relativeToComponentsPath(codegenConfig.getGeneratedSubscriptionsPath(), context),
+          fragmentsFilePath: relativeToComponentsPath(codegenConfig.getGeneratedFragmentsPath(), context),
+        },
+      };
+
+      const minQueryDepth = 3;
+      const isQueryingTooShallow = (codegenConfig.getQueryMaxDepth() || 0) < minQueryDepth;
+
+      if (studioClient.metadata.formFeatureFlags.isRelationshipSupported && isQueryingTooShallow) {
+        promptForUpdateCodegen = true;
+        printer.warn(`Forms with relationships require a maximum query depth of at least ${minQueryDepth}.`);
+      }
+      return apiConfiguration;
+    } catch {
+      promptForUpdateCodegen = true;
+      printer.warn(
```

Comment:
Why print a warning and not fail fast here?

Reply:
we chose to warn rather than fail fast as failing as an API configuration is not a hard requirement for every component a customer could be generating.

Reply:
I think either the warning's content is too alarming as it begins with "Error" or this is an inadequate mechanism to alert customers that a problem occurred they need to address.

Is there a limitation that's preventing us from determining if a component is expected from having an API configuration?